joern-cli/frontends/jimple2cpg/src/main/scala/io/joern/jimple2cpg/Jimple2Cpg.scala

```diff
@@ -116,6 +113,26 @@ class Jimple2Cpg {
     }
   }
 
+  private def loadSourceFiles(
+      sourceCodePath: String,
+      sourceFileExtensions: Set[String],
+      archives: List[String]
+  ) = {
+    (archives
+      .map(new ZipFile(_))
+      .flatMap(x => {
```

Comment:
This method is quite hard to read, partly because it does many things, partly because it's quite nested. A few ideas:
* extract into smaller methods
* introduce named helper variables
* rewrite map/flatMap into for comprehension
* less parens: `.flatMap(x => {...})` -> `.flatMap { x => ... }`

Reply:
Yeah I realise this method has become a bit of a mess as edge cases have been coded for on top of this method over time. I'll take some time to fix this up